### PR TITLE
Rdcc 3689

### PR DESCRIPTION
**Pull request by ManishParyani**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-3689

### Change description ###
Case Flags Api

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -14,13 +14,39 @@ management:
...
+  security:
+      oauth2:
+        client:
+          provider:
+            oidc:
+              issuer-uri: ${OPEN_ID_API_BASE_URI:https://idam-web-public.aat.platform.hmcts.net/o}
+          registration:
+            oidc:
+              client-id: microservicekey-rd-commondata-api
+              client-secret: ${COMMON_DATA_API_S2S_SECRET:}
```
Comment (SujeethKumarAri) [inline src/main/resources/application.yaml:49]: Should not this be `COMMONDATA_API_S2S_SECRET`?

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -64,3 +90,21 @@ security:
     - "/csrf"
     - "/error"
     - "/favicon.ico"
+
+idam:
+  s2s-auth:
+    totp_secret: ${COMMON_DATA_API_S2S_SECRET:}
```
Comment (SujeethKumarAri) [inline src/main/resources/application.yaml:96]: Should not this be `COMMONDATA_API_S2S_SECRET`?

**Review by SujeethKumarAri — CHANGES_REQUESTED** [review]

File: src/main/resources/application.yaml (hunk 2 above)
Comment (ManishParyani) [reply src/main/resources/application.yaml]: Thanks @SujeethKumarAri , Good Catch.
Fixed it.

File: src/main/resources/application.yaml (hunk 1 above)
Comment (ManishParyani) [reply src/main/resources/application.yaml]: Thanks @SujeethKumarAri , Good Catch.
Fixed it.

File: src/main/java/uk/gov/hmcts/reform/cdapi/controllers/CaseFlagApiController.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            message = "Forbidden Error: Access denied"
+        ),
+        @ApiResponse(
+            code = 500,
+            message = "Internal Server Error"
+        )
+    })
+    @GetMapping(
+        produces = APPLICATION_JSON_VALUE,
+        path = "/caseflags/service-id={service-id}"
```
[the remark this answers is not in the document]
Comment (ManishParyani) [reply src/main/java/uk/gov/hmcts/reform/cdapi/controllers/CaseFlagApiController.java:63]: Yeah i have raised this with Candy, We will check with @kavithagade12  as well.

File: src/functionalTest/java/uk/gov/hmcts/reform/cdapi/idam/IdamOpenIdClient.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+
+        log.info("openIdTokenResponse createUser response: " + createdUserResponse.getStatusCode());
+
+        assertThat(createdUserResponse.getStatusCode()).isEqualTo(201);
+
+        Map<String, String> userCreds = new HashMap<>();
+        userCreds.put(EMAIL, userEmail);
+        userCreds.put(CREDS, password);
+        return userCreds;
```
[the remark this answers is not in the document]
Comment (ManishParyani) [reply src/functionalTest/java/uk/gov/hmcts/reform/cdapi/idam/IdamOpenIdClient.java:83]: Thanks @SujeethKumarAri  for the reference, I have added the logic to delete users. 
I think this logic is missing in JRD and LRD Api's.

File: charts/rd-commondata-api/values.yaml [hunk 5]
```diff
@@ -10,12 +10,12 @@ java:
...
-    #OIDC_ISSUER_URL: https://forgerock-am.service.core-compute-idam-{{ .Values.global.environment }}.internal:8443/openam/oauth2/hmcts
-    #OPEN_ID_API_BASE_URI: https://idam-web-public.{{ .Values.global.environment }}.platform.hmcts.net/o
-    #IDAM_URL: https://idam-api.{{ .Values.global.environment }}.platform.hmcts.net
-    #LOCATIONS_DB: classpath:db/migration,classpath:db/postgres
-    #LAUNCH_DARKLY_ENV: "{{ .Values.global.environment }}"
+    S2S_URL: http://rpe-service-auth-provider-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal
+    OIDC_ISSUER_URL: https://forgerock-am.service.core-compute-idam-{{ .Values.global.environment }}.internal:8443/openam/oauth2/hmcts
+    OPEN_ID_API_BASE_URI: https://idam-web-public.{{ .Values.global.environment }}.platform.hmcts.net/o
+    IDAM_URL: https://idam-api.{{ .Values.global.environment }}.platform.hmcts.net
+    LOCATIONS_DB: classpath:db/migration,classpath:db/postgres
```
[the remark this answers is not in the document]
Comment (ManishParyani) [reply charts/rd-commondata-api/values.yaml]: If you are talking about LOCATIONS_DB then yes i have corrected this variable.

[the remark this answers is not in the document]
Comment (ManishParyani) [reply charts/rd-commondata-api/values.yaml]: Fixed

**Review by SujeethKumarAri — APPROVED** [review]

**Review by sahitya-desireddy — APPROVED** [review]

**Review by kavithagade12 — APPROVED** [review]

**Pull request merged**